twine/repository.py

```diff
@@ -37,14 +37,28 @@
 logger = logging.getLogger(__name__)
 
 
-class ProgressBar(tqdm.tqdm):
-    def update_to(self, n: int) -> None:
-        """Update the bar in the way compatible with requests-toolbelt.
+class CondensedTimeColumn(rich.progress.ProgressColumn):
+    """Renders estimated time remaining, or elapsed time when the task is finished."""
 
-        This is identical to tqdm.update, except ``n`` will be the current
-        value - not the delta as tqdm expects.
-        """
-        self.update(n - self.n)  # will also do self.n = n
+    # Only refresh twice a second to prevent jitter
+    max_refresh = 0.5
+
+    def render(self, task: rich.progress.Task) -> rich.text.Text:
+        """Show time."""
+        style = "progress.elapsed" if task.finished else "progress.remaining"
+        task_time = task.finished_time if task.finished else task.time_remaining
+        if task_time is None:
+            return rich.text.Text("--:--", style=style)
+
+        # Based on https://github.com/tqdm/tqdm/blob/master/tqdm/std.py
+        minutes, seconds = divmod(int(task_time), 60)
+        hours, minutes = divmod(minutes, 60)
+        if hours:
+            formatted = f"{hours:d}:{minutes:02d}:{seconds:02d}"
+        else:
+            formatted = f"{minutes:02d}:{seconds:02d}"
+
+        return rich.text.Text(formatted, style=style)
```

Comment:
Following up on https://github.com/pypa/twine/pull/877#discussion_r811127244.

After looking at several download progress indicators (e.g. Chrome, Firefox, `wget`, `scp`), it seems more common to show time remaining than time elapsed. Both `wget` and `scp` show time remaining until the download is finished, then switch to time elapsed. So, I implemented that behavior here.

Yes, this is bike-shedding/yak-shaving, but it was fun.

https://user-images.githubusercontent.com/1326704/155126794-0328b8a6-33ac-4d68-9482-607aada22aa4.mp4

Reply:
I also opened PR on Rich: https://github.com/Textualize/rich/pull/1992

Reply:
The PR was merged; I'll finish this once it's released.